### PR TITLE
Flashbang nerfs, grenade priming animation.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
- tweak: "Flashbangs now no longer stun you if you have moderate ear protection (aka a helmet)."
  - tweak: "Flashbangs now damage you quite a bit if they explode while in your hand or in your pockets. Be careful!"
  - tweak: "The BANG message after being stunned by a flashbang is now bigger."
  - tweak: "Grenades now get redder as they're about to explode. The timing is slightly randomized, so don't use it as an exact indicator of when to throw a grenade."

https://github.com/Aurorastation/Aurora.3/assets/13016572/7183d047-38b3-4b60-bec2-4911e17eede9

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

File: code/game/objects/items/weapons/grenades/flashbang.dm [hunk 1]
```diff
@@ -6,11 +6,30 @@
 
 /obj/item/grenade/flashbang/prime()
 	..()
+	/// Holding a flashbang when it goes off is bad news.
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/grenades/flashbang.dm:9]: These aren't DMDocs, double slash only goes here, DMDocs are just for vars and procs

```suggestion
	// Holding a flashbang when it goes off is bad news.
```

File: code/game/objects/items/weapons/grenades/flashbang.dm [hunk 2]
```diff
@@ -6,11 +6,30 @@
...
+			exploded_hand = victim.organs_by_name[BP_R_HAND]
+		else if(victim.l_hand == src)
+			exploded_hand = victim.organs_by_name[BP_L_HAND]
+		/// The flashbang was held in their hand, and we have one.
+		if(exploded_hand)
+			to_chat(victim, SPAN_HIGHDANGER("\The [src] goes off in your hand!"))
+			victim.apply_damage(30, DAMAGE_BRUTE, exploded_hand, src, DAMAGE_FLAG_EXPLODE, 20)
+			victim.apply_damage(20, DAMAGE_BURN, exploded_hand, src, DAMAGE_FLAG_EXPLODE, 20)
+		else
+			/// No hand, so it's in a bag or pocket.
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/grenades/flashbang.dm:23]: Same

File: code/game/objects/items/weapons/grenades/flashbang.dm [hunk 3]
```diff
@@ -6,11 +6,30 @@
...
+			to_chat(victim, SPAN_HIGHDANGER("\The [src] goes off on you!"))
+			victim.apply_damage(20, DAMAGE_BRUTE, exploded_hand, src, DAMAGE_FLAG_EXPLODE|DAMAGE_FLAG_DISPERSED, 20)
+			victim.apply_damage(30, DAMAGE_BURN, exploded_hand, src, DAMAGE_FLAG_EXPLODE|DAMAGE_FLAG_DISPERSED, 20)
+
 	var/turf/T = get_turf(src)
 	for(var/mob/living/L in get_hearers_in_view(7, src))
 		bang(T, L)
 
-	// Damage blobs
+	/// Damage blobs.
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/grenades/flashbang.dm:32]: Same

File: code/game/objects/items/weapons/grenades/grenade.dm [hunk 4]
```diff
@@ -76,6 +76,9 @@
 	playsound(loc, activation_sound, 75, 1, -3)
 
 	addtimer(CALLBACK(src, PROC_REF(prime)), det_time)
+	/// Randomize the timing a bit. You wouldn't be aware of when exactly a grenade is going to pop.
+	/// Nor do we want people to instantly know when to throw a perfectly timed grenade.
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/grenades/grenade.dm:80]: Same

File: code/game/objects/items/weapons/grenades/flashbang.dm [hunk 5]
```diff
@@ -6,11 +6,30 @@
...
 	..()
+	/// Holding a flashbang when it goes off is bad news.
+	if(ishuman(loc))
+		var/mob/living/carbon/human/victim = loc
+		var/obj/item/organ/external/exploded_hand
+		if(victim.hand == src)
+			exploded_hand = victim.organs_by_name[BP_R_HAND]
+		else if(victim.l_hand == src)
+			exploded_hand = victim.organs_by_name[BP_L_HAND]
+		/// The flashbang was held in their hand, and we have one.
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/grenades/flashbang.dm:17]: Same

**Review by Geevies — APPROVED** [review]
might be worth adding a proc that determines which hand an item is held in, but could be considered out of scope for this one

**Review by FluffyGhoster — APPROVED** [review]

**Pull request merged**